### DISCUSSION TITLE
Possible Incomplete Records Example?

### DISCUSSION THREAD

**Discussion by antontolken** [colinhacks/zod#1651] [Q&A] [answered]
i'm learning about zod, and looking at this example it feels like there may be something missing? for example what is `UserStore`, and / or where is zod actually used in this example?
```ts
const userStore: UserStore = {};

userStore["77d2586b-9e8e-4ecf-8b21-ea7e0530eadd"] = {
  name: "Carlotta",
}; // passes

userStore["77d2586b-9e8e-4ecf-8b21-ea7e0530eadd"] = {
  whatever: "Ice cream sundae",
}; // TypeError
```
related docs link: https://zod.dev/?id=records

**Comment by JacobWeisenburger** [comment]
[Records](https://github.com/colinhacks/zod#records)

You are correct that this example is not very good. Feel free to submit a PR to improve it. Thanks.

**Reply by antontolken** [reply]
Thanks for the confirmation.

There is one small issue: since I am learning about zod, I am unsure of what the above example should be.

If I do figure it out then I'm happy to make a PR.

**Comment by JacobWeisenburger** [comment, marked as answer]
It probably should look something like this

```ts
const userSchema = z.object( { name: z.string() } )
const userStoreSchema = z.record( userSchema )

type UserStore = z.infer<typeof userStoreSchema>
// type UserStore = { [ x: string ]: { name: string } }

const userStore: UserStore = {}

userStore[ '77d2586b-9e8e-4ecf-8b21-ea7e0530eadd' ] = {
    name: 'Carlotta',
}
console.log( userStoreSchema.safeParse( userStore ).success ) // true

userStore[ '77d2586b-9e8e-4ecf-8b21-ea7e0530eadd' ] = {
    whatever: 'Ice cream sundae',
}
console.log( userStoreSchema.safeParse( userStore ).success ) // false
```

**Reply by messenjer** [reply]
Here is the PR : https://github.com/colinhacks/zod/pull/2579
